fsx/merge_test.go

```diff
@@ -19,7 +20,48 @@ func TestMergeFS(t *testing.T) {
 		"b":     &fstest.MapFile{},
 		"dir/d": &fstest.MapFile{},
 	}
-	m := Merge(a, b)
+	x := fstest.MapFS{
+		"x":     &fstest.MapFile{},
+		"dir/y": &fstest.MapFile{},
+	}
+
+	assert.NoError(t, fstest.TestFS(
+		Merge(a, b),
+		"a",
+		"b",
+		"dir",
+		"dir/c",
+		"dir/d",
+	))
+
+	assert.NoError(t, fstest.TestFS(
+		Merge(a, b, x),
+		"a",
+		"b",
+		"dir",
+		"dir/c",
+		"dir/d",
+		"dir/y",
+		"x",
+	))
+
+	assert.Error(t, fstest.TestFS(
+		mergefs.Merge(a, b),
+		"a",
+		"b",
+		"dir",
+		"dir/c",
+		"dir/d",
+	))
 
-	assert.NoError(t, fstest.TestFS(m, "a", "b", "dir", "dir/c", "dir/d"))
+	assert.Error(t, fstest.TestFS(
+		mergefs.Merge(mergefs.Merge(a, b), x),
+		"a",
+		"b",
+		"dir",
+		"dir/c",
+		"dir/d",
+		"dir/y",
+		"x",
+	))
```

Comment:
It took me way too long to see that this was not our code, but the other lib :facepalm:

Reply:
And it took me way too long to see you changed `assert.NoError` to `assert.Error` while I was trying to understand how you magically made the tests pass for `laher/mergefs` also 🤦